x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/SettingsPage/SettingFormRow.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+import {
+  EuiFormRow,
+  EuiFieldText,
+  EuiFieldNumber,
+  EuiDescribedFormGroup,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiCode,
+  EuiSpacer,
+  EuiIconTip
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { isValid } from '../../../../../../../../../../plugins/apm/common/runtime_types/agent_configuration/config_setting_definitions';
+import {
+  amountAndUnitToString,
+  amountAndUnitToObject
+} from '../../../../../../../../../../plugins/apm/common/runtime_types/agent_configuration/amount_and_unit';
+import { ConfigSettingDefinition } from '../../../../../../../../../../plugins/apm/common/runtime_types/agent_configuration/config_setting_definitions';
+import { SelectWithPlaceholder } from '../../../../../shared/SelectWithPlaceholder';
+
+function FormRow({
```

Comment:
I really like this function, don't you think it would be nice if we could re-use it in other forms, maybe place it in a shared location?

Reply:
I don't think we should re-use it - it'll just end up in an abstracted mess is my experience. But I think we could use the same approach elsewhere if it makes sense.